iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber_uds.inl

```diff
@@ -0,0 +1,145 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+
+#include "owl/kom/event_subscriber_uds.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventSubscriber<T, EventTransmission::UDS>::EventSubscriber(const ServiceIdentifier&,
+                                                                   const InstanceIdentifier& instance,
+                                                                   const EventIdentifier&) noexcept
+    : m_uds(iox::posix::UnixDomainSocket::create(instance, iox::posix::IpcChannelSide::SERVER)
+                .expect("Failed to create UNIX domain socket!"))
+{
+}
+
+template <typename T>
+inline EventSubscriber<T, EventTransmission::UDS>::~EventSubscriber() noexcept
+{
+    m_run = false;
+    m_thread.join();
+}
+
+template <typename T>
+inline void EventSubscriber<T, EventTransmission::UDS>::Subscribe(std::size_t) noexcept
+{
+    // Subscribe not supported with UDS
+}
+
+template <typename T>
+void EventSubscriber<T, EventTransmission::UDS>::Unsubscribe() noexcept
+{
+    // Unsubscribe not supported with UDS
+}
+
+template <typename T>
+template <typename Callable>
+core::Result<size_t> inline EventSubscriber<T, EventTransmission::UDS>::GetNewSamples(Callable&& callable,
+                                                                                      size_t maxNumberOfSamples)
+{
+    IOX_DISCARD_RESULT(maxNumberOfSamples);
+
+    core::Result<size_t> numberOfSamples{1};
+
+    auto samplePtr = std::make_unique<SampleType>();
+    SampleType& sample = *samplePtr;
+
+    std::string tempBuffer;
+
+    // Receive the first (up to) 4095 Bytes
+    m_uds.timedReceive(iox::units::Duration::fromSeconds(1))
+        .and_then([&](auto& msg) { tempBuffer.append(msg); })
+        .or_else([](auto&) { std::cerr << "Error occurred while receiving UNIX domain socket!" << std::endl; });
+
+    if (tempBuffer.size() < 15)
+    {
+        return 0;
+    }
+
+    const char* readPtr = tempBuffer.c_str();
+
+    // Deserialize the counter
+    std::memcpy(&sample.counter, readPtr, sizeof(sample.counter));
+    readPtr += sizeof(sample.counter);
+
+    // Deserialize the timestamp
+    int64_t sendTimestamp;
+    std::memcpy(&sendTimestamp, readPtr, sizeof(sendTimestamp));
+    readPtr += sizeof(sendTimestamp);
+
+    sample.sendTimestamp =
+        std::chrono::time_point<std::chrono::steady_clock>(std::chrono::duration<int64_t, std::nano>(sendTimestamp));
+
+    // Deserialize subPackets
+    std::memcpy(&sample.subPackets, readPtr, sizeof(sample.subPackets));
```

Comment:
This is evil. You serialized this package in a different order then the actual struct layout of `TimestampTopic`. So I was wondering if this isn't a bug since you skipped the `payloadSizeInBytes` and `data` member.

In my opinion this is not how one should implement serialization since the serialization/deserialization order is hidden deep inside the implementation and does not confirm to the layout. You also break the single responsibility principle with this.

I suggest the following. The type itself has the responsibility to serialize/deserialize itself since it knows best how to serialize itself.

Luckily we already have tested and working serialization building block in iceoryx at our disposal `cxx::Serialization`. You add two things to `TimestampTopic`.
1. Constructor
```cxx
TimestampTopic::TimestampTopic(const cxx::Serialization & serialized) noexcept {
  if ( !serialized.extract(counter, sendTimestamp, payloadSizeInBytes, data, subPackets) ) {
    LogError() << "Failed to deserialize";
  }
}
```
2. Conversion operator
```cxx
TimestampTopic::operator cxx::Serialization() const noexcept {
  return cxx::Serialization::create(counter, sendTimestamp, payloadSizeInBytes, data, subPackets);
}
```

Under the hood `cxx::Serialization` is using `cxx::convert::fromString` and `cxx::convert::toString` since they do not support `std::chrono::time_point<std::chrono::steady_clock>` you have to either turn the member in `int64_t` and convert it later when you need it our provide an implementation to convert it. So you would have to implement `convert::toString` and `convert::fromString` for this data type. The implementation could be stored in your example, the linker will find it there.

See `node_property.cpp` or other constructs which we send via our IpcChannel (unix domain socket) between roudi and application.

Reply:
@elBoberido and myself had [this discussion](https://github.com/eclipse-iceoryx/iceoryx/pull/1355#discussion_r870588856) two weeks ago and concluded that it would be two much effort to use `cxx::Serialization`.

Anyway, closing, as this file wil be moved to a separate repository.